hexrd/material.py

```diff
@@ -775,9 +785,9 @@ def _set_atominfo(self, v):
         if v.shape[1] == 4:
             self._atominfo = v
         else:
-            print("Improper syntax, array must be n x 4")
+            raise ValueError("Improper syntax, array must be n x 4")
 
-        return
+        self.update_structure_factor()
```

Comment:
Not related to your commit, but I noticed that the atominfo is not getting updated in the unitcell class, which is what actually calculates the structure factors.
I'll push a fix for this soon.